### DISCUSSION TITLE
Download the log.txt

### DISCUSSION THREAD

**Discussion by supernacheitor** [luc-github/ESP3D#815] [Questions]
Is there any way to download the log.txt file from the web interface? If it is not possible, how can I consult it to see the error it gives me when uploading a gcode file to the SD?

**Comment by luc-github** [comment]
what log.txt are you refering to ?

**Reply by supernacheitor** [reply]
In the source code:

#define LOG(string) { FS_FILE logfile = SPIFFS.open("/log.txt", "a+");logfile.print(string);logfile.close();}

**Reply by luc-github** [reply]
ok this is when you enable debug, but enable general debug when doing upload is not good idea it may generate timeout or data lost

when upload failed it generate error code in webui, it usualy tell you the error

what kind of upload are you doing ? what FW version ? what board?

**Reply by supernacheitor** [reply]
An gcode upload.
BigTreeTech skr 2
I've created a new thread for this question [link](https://github.com/luc-github/ESP3D/discussions/817).

Where can I find the error information in webui?

**Reply by luc-github** [reply]
you already show the error= no connection
can you share the output of `[ESP420]` in webui terminal ?

**Reply by supernacheitor** [reply]
I don't know where I can open the webui terminal. I only install ESP3D-2.1 in the ESP.

**Reply by luc-github** [reply]
![image](https://user-images.githubusercontent.com/8822552/194292488-2cad3c99-5e87-4b62-8b4e-342564be79f8.png)

**Reply by supernacheitor** [reply]
The interface shows me a window that says Error: No connection. But the command window is empty.
Is there a command to view the logs?

![imagen](https://user-images.githubusercontent.com/101657488/194298477-1cfb3249-f378-4efd-923d-946047ceeb63.png)

**Reply by luc-github** [reply]
did you read what I wrote ?
1 - there is no log unless you enable debug 
2 - the error is no connection 
3 - I need you to type `[ESP420]` in webUI terminal - the location where I put red line around - and you need to give me the output - so I can have more informations of your config - to see why you loose connection

**Reply by supernacheitor** [reply]
I get that information:


{"chip_id":"4642229","cpu":"160","freemem":"35.31 KB","SDK":"2.2.1(cfd48f3)","flash_size":"4.00 MB","update_size":"534.09 KB","spiffs_size":"1.86 MB","baud_rate":"921600","sleep_mode":"Modem","channel":"6","phy_mode":"11g","web_port":"80","data_port":"8888","hostname":"ESP3D","active_mode":"STA (BC:FF:4D:46:D5:B5)","connected_ssid":"sagemcom8DB0","connected_signal":"98%","ip_mode":"DHCP","ip":"192.168.0.18","gw":"192.168.0.1","msk":"255.255.255.0","dns":"212.166.211.4","disabled_mode":"AP (BE:FF:4D:46:D5:B5)","captive_portal":"Enabled","ssdp":"Enabled","netbios":"Disabled","mdns":"Enabled","web_update":"Enabled","pin recovery":"Disabled","autentication":"Disabled","target_fw":"Marlin","M117_output":"Enabled","Notifications":"Disabled","Serial_output":"Enabled","Websocket_output":"Enabled","TCP_output":"Enabled","fw":"2.1.3b0 ESP8266/8586"}
{"chip_id":"4642229","cpu":"160","freemem":"35.31 KB","SDK":"2.2.1(cfd48f3)","flash_size":"4.00 MB","update_size":"534.09 KB","spiffs_size":"1.86 MB","baud_rate":"921600","sleep_mode":"Modem","channel":"6","phy_mode":"11g","web_port":"80","data_port":"8888","hostname":"ESP3D","active_mode":"STA (BC:FF:4D:46:D5:B5)","connected_ssid":"sagemcom8DB0","connected_signal":"98%","ip_mode":"DHCP","ip":"192.168.0.18","gw":"192.168.0.1","msk":"255.255.255.0","dns":"212.166.211.4","disabled_mode":"AP (BE:FF:4D:46:D5:B5)","captive_portal":"Enabled","ssdp":"Enabled","netbios":"Disabled","mdns":"Enabled","web_update":"Enabled","pin recovery":"Disabled","autentication":"Disabled","target_fw":"Marlin","M117_output":"Enabled","Notifications":"Disabled","Serial_output":"Enabled","Websocket_output":"Enabled","TCP_output":"Enabled","fw":"2.1.3b0 ESP8266/8586"}

**Reply by luc-github** [reply]
I see nothing - signal is ok - freq is 160 so ok 
I am only guessing a crash happen , a power supply issue ? esp boards are very sensitive 
how your esp is connected to printer ? is that a socket esp07 or you did the wire by yourself ?

**Reply by supernacheitor** [reply]
BigTreeTech MKR 2 has his own socket for the ESP.
![imagen](https://user-images.githubusercontent.com/101657488/194307811-c7d9db9a-ec18-495e-b52c-edff1919e1fc.png)
This is the ESP:
![imagen](https://user-images.githubusercontent.com/101657488/194308574-c31900c7-ab85-4317-96c3-d4c89e8f0dc0.png)

**Reply by luc-github** [reply]
Are you powered printer board only using usb ? or using external power supply
Does the board has also a screen ? 

The wifi upload may drain a lot of power at the begining and generate instability if only using USB power

**Reply by supernacheitor** [reply]
I use an external power supply, and the board has a screen.

**Reply by luc-github** [reply]
So I have no idea - does WebUI still communicate with printer immediatly after error pop up ?
it may help to know if the esp really crash or the upload failed and stop the connection in state without sending proper error message

**Reply by supernacheitor** [reply]
I don't know where the problem was, but today I received more wifi modules. When I changed the module for a new one, everything started to work.
Thank you very much for your help and sorry for the inconvenience.

**Reply by luc-github** [reply]
Hardware issue happen - thank you for the feedback

**Comment by supernacheitor** [comment]
One question, there is no way to send a file to the ESP through Wi-Fi, and have it printed directly. I use an ESP-07s that has 32Mbits of flash size, which is enough for all the gcode files that I usually print. The file could be stored in flash and the gcode instructions serially sent to the board while they are being printed. Could it be possible or am I wrong about something?

Thank you so much for everything.

**Reply by luc-github** [reply]
Let me try to answer:
* 32 Mbits => 4MBytes => on ESP8266 need 2 X 1MB for FW update => left 2MB to host files,  which is very small for 3D printing unlike you do very small and very simple objects
* Use Flash to serve file is bad idea, the flash has limited number of usage (max 100K to 10K times) and depending the quality of the flash the number can  still drop to few thousand very easily.
* The current GCode host stream is not yet ready for 3D printer (https://github.com/luc-github/ESP3D/issues/243) , no integrity check , no error checking and no resend,  you can try and  pray it works - it is planned but see next point as you use esp8266
* the esp8266 is one core chip  and does not handle several threads at once, which in case of GCODE streaming can be an issue if you access to webUI when doing streaming - you may have some latency issue and so some blob in final object if printer does not get commands on times.
So even when GCODE Host stream will be ready - I won't recommend it on esp8266 neither mono core, and on flash even less because when the webserver will server will serve the webui the gcode streaming will be out of control and so some latency will happen for sure.

ESP8266 is not a PC system, neither a raspberry PI -do not expect too much from a 1$ chip ^_^

After you can give a try to ESP3D V3 and hope 🙏   everything is going ok 😸

I hope it clarify things

**Reply by supernacheitor** [reply]
OK thank you very much. Everything is very clear to me.

In ESP3D v3, is there any functionality to print directly with ESP32 processors? I have seen in the configurator that you can select an ESP32-CAM board, which in addition to having its own SD card, has a 4MB PSRAM memory and a camera, of course (I also have several ESP32-CAMs from other projects).

Other options that I am considering is to use a board with an esp32 controller, such as FYSETC E4. But I have always worked with MTB boards.

**Reply by luc-github** [reply]
ESP3D V3 support ESP32 yes, better than 2.1

GCode stream status in V 3 is the one I gave you, ESP3D V 2.1 do not stream at all.

what is MBT Board sorry ?

**Reply by supernacheitor** [reply]
Sorry BTT, bigtreetech.
